src/bulkQueries.ts

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs/promises';
+import * as path from 'path';
+import { Calculator, DRCalculationOutput } from 'fqm-execution';
+
+/**
+ * Loads file from cli input to calculate bulkQueries from the file
+ */
+export async function cliBulkQueries(filePath: string) {
+  // Read in bundle
+  let data: string;
+  try {
+    data = await fs.readFile(path.resolve(filePath), 'utf8');
+  } catch (err) {
+    console.error('Error reading the bundle: ', err);
+    return;
+  }
+
+  const bundle = JSON.parse(data) as fhir4.Bundle;
+  const queries = await bulkQueries(bundle);
+  console.log(JSON.stringify(queries));
+}
+
+/**
+ * Generates the queries that would come after a bulk "$export?""
+ */
+export async function bulkQueries(bundle: fhir4.Bundle) {
+  // simple approach...
+  // If there's a codeFilter with a direct code, create a typeFilter for that code
+  // If the codeFilter doesn't have a code (valueset or some other specification), create a generic _type query instead
+  // This _type query should supercede all other potential narrowing that's found on that resource
+  // If we can narrow:
+  // Within the same DR codeFilter list, typeFilters should be ANDed (i.e. _typeFilter=A,B)
+  // Across different DR's, typeFilters should be ORd (i.e.  _typeFilter=A&_typeFilter=B)
+  // Make sure to also include a _type query for all resource types needed
+
+  const dataRequirements: DRCalculationOutput = await Calculator.calculateDataRequirements(bundle);
+
+  // create record resourcetype => [] of valid typeFilter query strings that will be &-ed as ORs
```

Comment:
```suggestion
  // create record resourcetype => [] of valid typeFilter query strings that will be separated by "&" and logically handled as ORs
```
or something perhaps a little clearer?